apps/heft/src/plugins/JestPlugin/jestWorkerPatch.ts

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fs from 'fs';
+import * as path from 'path';
+import { Import } from '@rushstack/node-core-library';
+
+// This patch is a fix for a problem where Jest reports this error spuriously on a machine that is under heavy load:
+//
+// "A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests
+// leaking due to improper teardown. Try running with --runInBand --detectOpenHandles to find leaks."
+//
+// The upstream issue is here: https://github.com/facebook/jest/issues/11354
+//
+// The relevant code is in jest-worker/src/base/BaseWorkerPool.ts:
+// https://github.com/facebook/jest/blob/64d5983d20a628d68644a3a4cd0f510dc304805a/packages/jest-worker/src/base/BaseWorkerPool.ts#L110
+//
+//      // Schedule a force exit in case worker fails to exit gracefully so
+//      // await worker.waitForExit() never takes longer than FORCE_EXIT_DELAY
+//      let forceExited = false;
+//      const forceExitTimeout = setTimeout(() => {
+//        worker.forceExit();
+//        forceExited = true;
+//      }, FORCE_EXIT_DELAY);
+//
+// The problem is that Jest hardwires FORCE_EXIT_DELAY to be 500 ms.  On a machine that is under heavy load,
+// the IPC message is not received from the child process before the timeout elapses.  The mitigation is to
+// increase the delay.  (Jest itself seems to be a significant contributor to machine load, so perhaps reducing
+// Jest's parallelism could also help.)
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type BaseWorkerPoolModule = any;
+
+// Follow the NPM dependency chain to find the module path for BaseWorkerPool.js
+// heft --> @jest/core --> @jest/reporters --> jest-worker
+
+const PATCHED_FORCE_EXIT_DELAY: number = 7000; // milliseconds
+
+try {
+  let contextFolder: string = __dirname;
+  contextFolder = Import.resolvePackage({ packageName: '@jest/core', baseFolderPath: contextFolder });
+  contextFolder = Import.resolvePackage({ packageName: '@jest/reporters', baseFolderPath: contextFolder });
+  const jestWorkerFolder: string = Import.resolvePackage({
+    packageName: 'jest-worker',
+    baseFolderPath: contextFolder
+  });
+
+  const baseWorkerPoolPath: string = path.join(jestWorkerFolder, 'build/base/BaseWorkerPool.js');
+
+  if (!fs.existsSync(baseWorkerPoolPath)) {
+    throw new Error(
+      'The BaseWorkerPool.js file was not found in the expected location:\n' + baseWorkerPoolPath
+    );
+  }
+
+  // Load the module
+  const baseWorkerPoolModule: BaseWorkerPoolModule = require(baseWorkerPoolPath);
+
+  // Obtain the metadata for the module
+  const baseWorkerPoolModuleMetadata: NodeModule = module.children[module.children.length - 1];
```

Comment:
This is nasty. Can you get the child by name? What happens if this code gets moved around?

Reply:
The children are appended in the order that they were `require()`'d.  The lookup key is the absolute file path, whose equality is difficult to compute.

I suppose I can search the list for a matching filename.